src/net.cpp

```diff
@@ -876,16 +855,16 @@ static bool AttemptToEvictConnection(bool fPreferNewConnection) {
     std::vector<unsigned char> naMostConnections;
     unsigned int nMostConnections = 0;
     int64_t nMostConnectionsTime = 0;
-    std::map<std::vector<unsigned char>, std::vector<CNodeRef> > mapAddrCounts;
-    for (const CNodeRef &node : vEvictionCandidates) {
-        mapAddrCounts[node->addr.GetGroup()].push_back(node);
-        int64_t grouptime = mapAddrCounts[node->addr.GetGroup()][0]->nTimeConnected;
-        size_t groupsize = mapAddrCounts[node->addr.GetGroup()].size();
+    std::map<std::vector<unsigned char>, std::vector<NodeEvictionCandidate> > mapAddrCounts;
+    BOOST_FOREACH(const NodeEvictionCandidate &node, vEvictionCandidates) {
```

Comment:
Take care to not revert these kinds of cleanups when cherry-picking (the original commit was prior to `BOOST_FOREACH` removal). Squash this revert into 0a369a5091b799332ed6e26e53d05dd8181e1ec1 (so it lines up more closely with the original commit):
```suggestion
    for (const NodeEvictionCandidate &node, vEvictionCandidates) {
```

Reply:
This is not yet resolved.